### PR TITLE
Added article Using Subqueries to Avoid the Eager

### PR DESCRIPTION
**Pull request by InverseFalcon**

### CODE REVIEW

File: articles/modules/ROOT/pages/using-subqueries-to-avoid-the-eager.adoc [hunk 1]
```diff
@@ -0,0 +1,130 @@
+= Using Subqueries to Avoid the Eager
+
```
Comment (Mogztter) [inline articles/modules/ROOT/pages/using-subqueries-to-avoid-the-eager.adoc:2]: you should remove this blank line:

```suggestion
```

File: articles/modules/ROOT/pages/using-subqueries-to-avoid-the-eager.adoc [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+:slug: using-subqueries-to-avoid-the-eager
+:author: Andrew Bowman
+:neo4j-versions: 4.1, 3.5
```
Comment (Mogztter) [inline articles/modules/ROOT/pages/using-subqueries-to-avoid-the-eager.adoc:5]: Version 4.1 does not exist on WordPress, should we create it? I can do it but I need a confirmation from the KB team.

Comment (InverseFalcon) [reply articles/modules/ROOT/pages/using-subqueries-to-avoid-the-eager.adoc:5]: Sorry, was that created? We will need entries for each minor version, so we will need 4.1 and 4.2.

**Comment by InverseFalcon** [comment]
Hold off on merging. The CLG seems intent on "fixing" this as they see this as buggy behavior. I'm trying to fight them off.

**Comment by InverseFalcon** [comment]
Abandoning this one, CLG decided to fix this workaround in Neo4j 4.3.x, so this article only documents the "buggy" behavior in versions below that.

Also, in discussions with some of the Cypher team, they pointed out that this workaround is not complete. There are other operators that operate in an eager-like way (hash join operators, PIPELINED runtime, parallel runtime, others) that are not covered by this workaround. 

For the future, we will be looking at official language constructs to denote segments in the query that should be free of any eager-like behavior (guaranteeing execution-per-row semantics, which go against normal Cypher semantics).

**Pull request closed without merging**